packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -373,6 +373,12 @@ export const RemoteServerFactory = {
                 console.log(`Req ${res.locals.requestId} for x-edge-request-id ${cloudfrontId}`)
             }
 
+            // if the request has x-amzn-requestid, assign it to x-correlation-id
+            const amzRequestId = req.headers['x-amzn-requestid']
+            if (amzRequestId) {
+                req.headers['x-correlation-id'] = amzRequestId
```

Comment:
`req.headers` is often treated as a readonly property. it is not supposed to be mutated, it should only contain the original data for the request.

Reply:
is it possible to have a undefined `req.headers['x-amzn-requestid']`?

Reply:
Do we want to have a response header for correlation id?